maven-metadata-inferer/src/main/java/org/ndx/aadarchi/inferer/maven/enhancers/SoftwareSystemEnhancer.java

```diff
@@ -26,8 +28,33 @@ protected Container getContainedElementWithName(String key) {
 	}
 
 	@Override
-	protected void containedDependsUpon(Container contained, Container found, String string) {
-		contained.uses(found, string);
+	protected String changeRelationshipDescription(Relationship uses) {
+		String relationshipDescription = null;
+		RelationshipsPropertiesFileReader propertiesFileReader = new RelationshipsPropertiesFileReader();
+		Map<String, String> relationshipDescriptionProperties = propertiesFileReader.readPropertiesFile("base/src/architecture/resources/relationships-description.properties");
```

Comment:
OK, here is the heart of the misunderstanding.

The relationship names will be *in* the project we want to document, not in the `base` module (which is used by everybody). As a consequence, the code in base (which should be common to maven, npm, and so on) should be able to "fix" the relationship description and expose an API such as

```
public interface RelationshipRenamed {
    /**
     * If possible, replace the relationship empty name with one read in the given property file
     */
    public Relationship nameDependencyRelationship(Relationship toRename);
}
```